command/agent/config.go

```diff
@@ -1089,6 +1093,57 @@ func (n *NormalizedAddrs) Copy() *NormalizedAddrs {
 	return &nn
 }
 
+// SocketConfig contains the path and configuration for a unix domain socket
+// listener.
+type SocketConfig struct {
+	Path  string `hcl:"path"`
+	User  string `hcl:"user"`
+	Mode  string `hcl:"mode"`
+	Group string `hcl:"group"`
```

Comment:
Some prior art: 
- [Vault](https://developer.hashicorp.com/vault/docs/configuration/listener/unix) has the bare minimum docs here. 
- [Consul](https://developer.hashicorp.com/consul/docs/agent/config/config-files#unix_sockets) is probably a good one to copy (except for the lack of Windows support... I wonder if it just-started-working for them when Go+Windows added support).
- [Boundary](https://developer.hashicorp.com/boundary/docs/configuration/listener/unix) is clearly an iteration on Vault's approach, and I don't think has anything relevant for us. It's interesting they appear to enable tls by default while Vault totally lacks tls for their unix socket. I think we can follow Vault here until somebody gives us a reason to do otherwise.